core/src/main/java/org/apache/iceberg/TableProperties.java

```diff
@@ -134,4 +134,10 @@ private TableProperties() {
 
   public static final String ENGINE_HIVE_ENABLED = "engine.hive.enabled";
   public static final boolean ENGINE_HIVE_ENABLED_DEFAULT = false;
+
+  public static final String ROW_LEVEL_OPS_ISOLATION_LEVEL = "row-level-ops.isolation-level";
+  public static final String ROW_LEVEL_OPS_ISOLATION_LEVEL_DEFAULT = "serializable";
```

Comment:
What about `write.spark.isolation-level`? And if we want to be more specific, we could use `write.spark.delete-from.isolation-level`.

Reply:
Do we want to make this property specific to Spark? Seems generic enough and I hope it will be respected by other query engines in the future too.

If we want to keep it Spark specific, it is going to be the first Spark table property. The only other query engine specific property is `engine.hive.enabled`. Do we want to share the `engine` prefix in this case?

Reply:
I'd probably group this in `write` because it is a setting that affects write behavior. If we don't want it to be specific to Spark, then we can make it `write.isolation.default-level` maybe?

I'm okay not making it specific to Spark, but it may be something that not every engine respects. Are we okay with that?